packages/test-runner-selenium/src/seleniumLauncher.ts

```diff
@@ -96,6 +102,25 @@ export class SeleniumLauncher implements BrowserLauncher {
 
     return this.iframeManager;
   }
+
+  private heartbeat() {
+    const driver = this.driver;
+
+    // Heartbeat function to keep alive sessions on Sauce Labs via WebDriver calls
+    this.pendingHeartbeat = setTimeout(async () => {
+      if (!driver) {
```

Comment:
I would query the driver from `this` fresh each time in the setTimeout callback. This way we properly pick up reassignment of the property.

Reply:
Done.